functions/src/api/middlewares/authenticateJWTMiddleware.ts

```diff
@@ -0,0 +1,57 @@
+import { Request, Response } from 'express'
+import * as jsonWebToken from 'jsonwebtoken'
+import { getJWTEnv } from '../../firebase/assertConfig'
+import { JWTData, JWTToken } from '../../types/JWT'
+import {
+    ForbiddenError,
+    PreconditionFailedError,
+    UnauthorizedError,
+} from '../errors/HttpError'
+import { assertValidToken } from '../auth/assertValidToken'
+
+export const authenticateJWTMiddleware = (
+    req: Request,
+    res: Response,
+    next: Function
+) => {
+    const jwtKey = getJWTEnv()
+
+    const authHeader = req.headers.authorization
+
+    if (!authHeader) {
+        throw new UnauthorizedError('No authorization header')
+    }
+
+    const splittedHeader = authHeader.split(' ')
+
+    if (splittedHeader.length < 2) {
+        throw new PreconditionFailedError('Malformed authorization header')
+    }
+
+    const token = <JWTToken>splittedHeader[1]
+
+    jsonWebToken.verify(
+        token,
+        jwtKey,
+        {
+            algorithms: ['HS256'],
+        },
+        async (err: jsonWebToken.VerifyErrors | null, data: {} | undefined) => {
```

Comment:
pls consider extracting this function from `authenticateJWTMiddleware`

Reply:
not agree, the code is small enough and explicit enough.

Reply:
let's leave it this way in this case. Function length is still a little subjective, but if I correctly recall, Uncle Bob describes clean code functions as having an optimal length of like 4 lines. I wouldn't be too strict on the subject either, but I thought that declaring it as a separate function would have reduced the vertical length (a measure for complexity) of the function (which is like around 40 lines now).

Reply:
thanks for the feedback, I'll keep in mind for the next PR!